app/models/user.rb

```diff
@@ -163,6 +163,14 @@ def internal?
     auth_source.is_a? AuthSourceInternal
   end
 
+  def ldap?
+    auth_source.is_a? AuthSourceLdap
+  end
+
+  def can_update_username?
+    !ldap?
```

Comment:
I think, it would be better to make this an option of the AuthSource. That way this would use polymorphism.

```ruby
delegate :can_update_username?, :to => :auth_source
```